source/ide/ide_filelock.h

```diff
@@ -0,0 +1,65 @@
+#include <stdint.h> // standard integer types
+
+#ifdef QB64_WINDOWS
+
+/// We wouldn't need to call these functions in Windows, but to avoid
+/// the need for platform-conditional checks on the BASIC side we simply
+/// define the functions as stubs in Windows.
+///-------------------------------------------------------------------
+int32_t LockGlobalFileAccess(const char *filename) {
+    return 0; // always succeeds
+}
+
+void ReleaseGlobalFileAccess(void) {
+    return;
+}
+
+#else // = QB64_LINUX or QB64_MACOSX
+
+/// The real functions used in Linux and MacOSX will either open and
+/// lock, or unlock and close the file respectively.
+///-------------------------------------------------------------------
+#include <fcntl.h>    // requred for open()
+#include <sys/file.h> // requred for flock()
+#include <unistd.h>   // requred for close()
+
+// Saved file descriptor, as this is global we can only lock one file
+// at a time, but that's sufficient for our use case.
+//--------------------------------------------------------------------
+int32_t filedesc = -1;
+
+// Try to lock the given file to prevent other IDE instances from
+// accessing the same file at the same time.
+//  In: filename (STRING, add CHR$(0) to end of string)
+// Out: error    (INTEGER, 0 = no error (file locked), -1 = error (file not locked))
+//--------------------------------------------------------------------
+int32_t LockGlobalFileAccess(const char *filename) {
+    int32_t err = -1;     // so far assume error
+    if (filedesc == -1) { // make sure only one file at a time
+        filedesc = open(filename, O_RDWR);
```

Comment:
I think we should differentiate the `open()` failing and the `flock()` failing. If the `open()` fails due to something like a permissions issue then we'll want to tell the user about that (`UpdateHistory()` would get stuck in a loop).

Reply:
I'll chack that tomorrow.

Reply:
Well this one...
It's a use case scenario here, no intention to create something commonly usable. Thinking about the if and when those functions are called (and I always do such logic assumptions when coding, case it can save a lot of work) I see no reason why the open should fail at all, so to say I could just remove the open() fail check completely. The path alraedy exists and must be accessible, otherwise the none of the files could be read. The file I try to lock does already exist (the signal file), so we can assume if something would prevent accessing/writing the file that would already happen long before trying to open/lock the file. All we could do is to catch it and telling the user to check and possibly whitelist his qb64pe folder, but as we already do mention that in the Wiki, Forum, ReadMe... I don't currently see a need to add another such mention in code.